pywbem/tupleparse.py

```diff
@@ -92,6 +91,21 @@
 from .exceptions import CIMXMLParseError
 from ._warnings import ToleratedServerIssueWarning
 
+# test to separate python 2 from python 3 for getting string types. Alternate
+# to using six.string_types and six.text
+try:
+    # attempt to evaluate basestring
+    basestring  # pylint: disable=pointless-statement
+    STRING_TYPES = basestring
+except NameError:
+    STRING_TYPES = str
+
+try:
+    # attempt to evaluate unicode
+    unicode  # pylint: disable=pointless-statement
+    TEXT_TYPE = unicode
+except NameError:
+    TEXT_TYPE = str
```

Comment:
I don't understand the reason for this change. It boils down to exactly the same types six provides. Please explain.

Reply:
Because at the time I had not put it up for review but just to test against CI.  The isinstance was the most expensive item in cprofile that we had control off so it seemed logical to see what could be done with it.  Actually you are just about correct the difference between using six and directly defining the types is alsmost nothing. The gain is in removing the number of isinstance calls since it was running at about 650,000 for our std test. and was #2 on the cprofile list

Reply:
Note that I completed the code and now there is no use of isinstance in tupleparse.py

Reply:
After discussing it in today's call and after understanding it better, the reason the `type(...) == six.string_types` did not work is that `six.string_types` is a tuple (with one item). So it should be `type(...) in six.string_types` instead. Could you please try with that, instead of using STRING_TYPE?

Reply:
Also, one of the uses of STRING_TYPE can be eliminated competely: In pcdata(), the entire for loop can be removed, and the join can be protected by handling TypeError and then raising the CIMXMLParseError.